evm/vm/base.py

```diff
@@ -133,45 +133,48 @@ def execute_bytecode(self,
     #
     def import_block(self, block):
         self.receipts = []
-        self.configure_header(
+        self.block = self.block.copy(header=self.configure_header(
             coinbase=block.header.coinbase,
             gas_limit=block.header.gas_limit,
             timestamp=block.header.timestamp,
             extra_data=block.header.extra_data,
             mix_hash=block.header.mix_hash,
             nonce=block.header.nonce,
             uncles_hash=keccak(rlp.encode(block.uncles)),
-        )
+        ))
 
         # run all of the transactions.
         for transaction in block.transactions:
             self.apply_transaction(transaction)
 
         # transfer the list of uncles.
-        self.block.uncles = block.uncles
+        self.block = self.block.copy(
+            uncles=block.uncles,
+            transactions=block.transactions,
```

Comment:
It feels wrong to overwrite `transactions` immediately after we've applied those transactions to the block.

Reply:
Probably a mistake, I'll revisit.